app/main.py

```diff
@@ -87,128 +1,19 @@
+from app.battle import Battle
+from app.knight.factory import get_knights
+from app.config import KNIGHTS
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def battle(knights_config: dict) -> dict:
+    knights = get_knights(knights_config)
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    first_battle = Battle()
+    result1 = first_battle.fight(knights[0], knights[2])
 
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
+    second_battle = Battle()
+    result2 = second_battle.fight(knights[1], knights[3])
 
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
+    return {**result1, **result2}
```

Comment:
There is a repetition in your code when creating `Battle` instances and calling the `fight` method. This violates the DRY (Don't Repeat Yourself) principle. Consider using a loop to create these instances and call the `fight` method. You can store the knight pairs in a list and iterate over them.